test/test_e2e.py

```diff
@@ -5,24 +5,29 @@
 import subprocess
 import pytest
 
-ABANDONING_PARENT = str(Path(__file__).with_name("data") / "abandoning_parent.sh")
+TEST_SCRIPT_DIR = Path(__file__).with_name("data")
 
 
 def test_sanity(temp_output_dir: str) -> None:
     command = f"duct -p {temp_output_dir}log_ sleep 0.1"
     subprocess.check_output(command, shell=True)
 
 
+@pytest.mark.parametrize("mode", ["plain", "subshell", "nohup", "setsid"])
 @pytest.mark.parametrize("num_children", [1, 2, 10])
-def test_abandoning_parent(temp_output_dir: str, num_children: int) -> None:
+def test_spawn_children(temp_output_dir: str, mode: str, num_children: int) -> None:
     duct_prefix = f"{temp_output_dir}log_"
-    command = f"duct --s-i 0.001 --r-i 0.01 -p {duct_prefix} {ABANDONING_PARENT} {num_children} sleep 0.2"
+    script_path = TEST_SCRIPT_DIR / "spawn_children.sh"
+    command = f"duct -q --s-i 0.001 --r-i 0.01 -p {duct_prefix} {script_path} {mode} {num_children} sleep 0.2"
     subprocess.check_output(command, shell=True)
 
     with open(f"{duct_prefix}usage.json") as usage_file:
         all_samples = [json.loads(line) for line in usage_file]
 
     all_pids = set(chain.from_iterable(sample["processes"] for sample in all_samples))
 
-    # 1 for each child, 1 for pstree, 1 for parent
-    assert len(all_pids) == num_children + 2
+    # Add 1 for the parent, add 1 for the "hold the door" process, see line 7 of the script
+    if mode == "setsid":
+        assert len(all_pids) == 2
```

Comment:
so we do not track/kill children correctly if underlying process (re)sets the sid, correct?

Reply:
I wonder if we need to indeed resort to the walking the tree of processes to figure them all out!  @vmdocua did similar in reprostim IIRC

Reply:
Correct that's a fundamental limitation

Reply:
https://github.com/con/duct/issues/259